src/NLog/Layouts/CompoundLayout.cs

```diff
@@ -104,5 +105,19 @@ protected override void CloseLayout()
                 layout.Close();
             base.CloseLayout();
         }
+
+        /// <summary>
+        /// Generate description of Compound Layout
+        /// </summary>
+        /// <returns>Compound Layout String Description</returns>
+        public override string ToString()
+        {
+            if (Layouts?.Count > 0)
+            {
+                var layoutDetails = Layouts.Select(l => l.ToString()).ToArray();
+                return string.Concat(GetType().Name, "=", string.Join("|", layoutDetails));
```

Comment:
Could we write an helper in the base class? 

I'm trying to prevent code duplication to keep things as consistent as possible.

Reply:
Have created an internal method for sub-classes to use, that sacrifices some more memory.